admin/worker/run_autoscaler.go

```diff
@@ -2,11 +2,23 @@ package worker
 
 import (
 	"context"
+	"time"
 
+	"github.com/rilldata/rill/admin/database"
 	"github.com/rilldata/rill/admin/metrics"
 	"go.uber.org/zap"
 )
 
+const legacyRecommendTime = 7 * 24 * time.Hour
+
+const scaleThreshold = 0.10
+
+func shouldScale(originSlots, recommendSlots int) bool {
+	lowerBound := float64(originSlots) * (1 - scaleThreshold)
+	upperBound := float64(originSlots) * (1 + scaleThreshold)
+	return float64(recommendSlots) < lowerBound || float64(recommendSlots) > upperBound
+}
```

Comment:
nit: put util functions at the end of the file. See the style guide on function ordering: https://github.com/uber-go/guide/blob/master/style.md#function-grouping-and-ordering